### ISSUE TITLE
Command line reports errors, language server does not

### ISSUE THREAD

**Issue by kyle-query** [microsoft/pyright#4780]
**Describe the bug**

Both neovim and Visual Studio code don't report errors, show no "unknown" types, and show the inferred types of values. However running pyright on the command line reports multiple unknown types (which should not be unknown).

**To Reproduce**

See [this small example](https://gist.github.com/kyle-query/a61bb8e14f96264f8d7e9f3e8f123c0e). I haven't narrowed the problem to OpenTelemetry, but I noticed the problem around the time I introduced this dependency in a project.

**Expected behavior**

The command line client should not report unknown types, as they are correctly inferred and not unknown or partially unknown by the language server.

**Screenshots or Code**

![Screen Shot 2023-03-15 at 4 19 36 PM](https://user-images.githubusercontent.com/108306629/225446714-eb194d4c-9a39-4454-b8ad-cda2d5a93eb8.png)

**VS Code extension or command-line**

Running pyright v1.1.299 on the command line and in neovim. Running pylance v2023.3.21 in Visual Studio Code.

**Additional context**

```
$ pyright --version
pyright 1.1.299
```

To rule out the issue of PYTHONPATH not being configured correctly by `poetry run`, I did confirm if I run `poetry run python test.py`, it prints "ok" and there are no errors.

**Comment by erictraut** [comment]
If you're getting different results between the language server and the command line, the most likely explanation is that you have something configured differently between the two. The type checker code is the same in both cases, so the behavior should be the same if configured the same.

Some things for you to investigate:
1. Do you have the same Python environment selected in both cases. You can tell by looking at the log, which will include the import resolution paths for your selected environment.
2. Do you have any language server settings set? These settings will be composed with any settings in your pyproject.toml (or pyrightconfig.json) file.
3. Do you have different versions of pyright installed (1.1.299 for the CLI and some older version for the language server)? The version should be logged in both cases.

**Comment by kyle-query** [comment]
Thanks Eric, I think I narrowed it down to "Use Library Code for Types". When I uncheck this in Visual Studio Code, I get similar type errors as I see on the command line. My problem seems to be resolved by setting useLibraryCodeForTypes to `true` in pyproject.toml. It looks like pyright is able to generate type stubs for libraries used in my project, so I'll look into that to avoid having to analyze the OpenTelemetry library source every time pyright runs.
